python/batch-predictor/docker/base.Dockerfile

```diff
@@ -12,44 +12,87 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM gcr.io/spark-operator/spark-py:v2.4.5
+ARG SPARK_VERSION=v3.0.0
+FROM gcr.io/spark-operator/spark-py:${SPARK_VERSION}
+
+# Switch to user root so we can add additional jars and configuration files.
+USER root
+
 ENV LANG=C.UTF-8 LC_ALL=C.UTF-8
 ENV PATH /opt/conda/bin:$PATH
-ENV ARROW_PRE_0_15_IPC_FORMAT 1
+
+ARG SPARK_OPERATOR_VERSION=v1beta2-1.3.7-3.1.1
```

Comment:
Are you planning to make SPARK_OPERATOR_VERSION and SPARK_BQ_CONNECTOR_VERSION configurable?

Reply:
nope, but this helps with upgrading efforts and helps prevent errors, since the values are used in more than 1 place.

Reply:
In that case I think you should use `ENV` instead of `ARG`